### DISCUSSION TITLE
Please advice me on best approach to creating (my first) journal template

### DISCUSSION THREAD

**Discussion by sorenoneill** [quarto-dev/quarto-cli#9265] [Q&A]
### Description

I want to try my hand at creating a template for a Springer Nature BMC journal (C&MT).

A LaTeX template is provided by the publishers [here](https://www.springernature.com/gp/authors/campaigns/latex-author-support). The publisher, admonishes authors not to stray far from the template, e.g. include niché packages etc.

The template includes commands for the author declaration like this:
````qmd
\author*[1,2]{\fnm{First} \sur{Author}}\email{iauthor@gmail.com}
````

You will notice, the `fnm` and `sur` functions which are particular to the template and different to the quarto normalized author format (afaik). 

My question is this:

In general terms, what would be the recommended way to go about implementing such things in a new quarto template?

1. Should I base my quarto markdown on the publishers template and use a `template-partial` title.tex 
2. ..or should I base my quarto markdown on Pandoc and use a `template-partial` title.tex, doc-class.tex, etc
3. ..or rely entirely on the publishers template, and modify it directly

Whether using the publishers template in its entirety or using `template-partials`, would you recommend implementing the `fnm` and `sur` functions:

1. as YAML variables and included as `$fnm$` in the tex partial or
2. hardcoded into the title.tex template - i.e simply editing the title.tex to reflect actual authors names etc

...I am looking mostly for hints on general approach and arguments why

**Comment by mcanouil** [comment]
I suggest to go to the documentation and follow it:
- https://quarto.org/docs/journals/formats.html

Note that the community already made a Quarto template extension for Nature, which you can find:
- https://quarto.org/docs/extensions/listing-journals.html

**Reply by sorenoneill** [reply]
Wauv. That was quick :-) 

I've already studied the documentation (of course) - what I was hoping for, was some reasoning behind choosing pandoc based over template based.

...the Nature template, I had missed however - thanks for the pointer. I'll try it out right away.

**Reply by mcanouil** [reply]
> was some reasoning behind choosing pandoc based over template based.

I don't follow. What do you mean by "pandoc based" and "template based"?

Quarto has an extension mechanism that allows users to create template extension.
A template extension can use a "template.tex" which is a whole template, or partials which are parts included in the default LaTeX template (originally embedded in Pandoc), see <https://quarto.org/docs/journals/templates.html#latex-partials>.

So, I don't understand what you mean.
Having one single file or partials is really up to you and what you want your extension users be able to do with it.

**Reply by sorenoneill** [reply]
Well, the docs page on [templates](https://quarto.org/docs/journals/templates.html) describes how "Quarto uses Pandoc templates to generate the rendered output", and that "by providing your own custom template used when rendering, you have complete control of the final output". It goes on to suggest "It’s therefore recommended that you take one of two approaches when authoring templates", which are described in greater detail.

I get your point however, and will start from there. Thanks for you support.
